### ISSUE TITLE
unknown variable not caught in `dict` constructor

### ISSUE THREAD

**Issue by ddfisher** [python/mypy#1357]
A file with just
```python
x = dict(undefined_variable)
```
succeeds, when it should give a `Name 'undefined_variable' is not defined` error.  Without the `dict` constructor, the error is generated correctly.

**Comment by ddfisher** [comment]
And here's why: https://github.com/python/mypy/blob/master/mypy/semanal.py#L1813

**Comment by gvanrossum** [comment]
Decided to fix this, in case it had a bearing on the more complex issue we were investigating yesterday (though I think it doesn't, actually).
